rust/src/nasl/interpreter/mod.rs

```
@@ -499,7 +499,14 @@ impl<'ctx> Interpreter<'ctx> {
     }
 
     pub(crate) async fn resolve_block(&mut self, block: &Block<Statement>) -> Result {
-        self.register.create_child();
+        let mut fncall = false;
+        if let Some(Statement::FnDecl(_)) = block.items.first() {
+            fncall = true;
+        }
+
+        if (fncall && self.version == NaslVersion::V1) || self.version == NaslVersion::V2 {
```

I recommend matching over versions instead of checking them in an if, so that we will automatically be notified of all the sites in the code we need to check if we add `NaslVersion::V3` at some point:

```
let add_scope = match self.version { 
    V1 => false,
    V2 => true,
};
```

or, also acceptable:

```
let add_scope = !matches!(self.version, NaslVersion::V1);
```

but not 

```
let add_scope = matches!(self.version, NaslVersion::V2);
```

since the latter would silently fail for (the hypothetical) V3